### PR TITLE
fix: aer mask lower bound

### PR DESCRIPTION
**Pull request by vishwa2710**
Taking the fixes from https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/606
Authored by litghost.

## Summary by CodeRabbit

* **Bug Fixes**
  * Normalize azimuth and elevation before mask lookups to ensure consistent geometric comparisons and correct boundary handling, fixing incorrect outcomes near wraparound and edge cases.

* **Tests**
  * Expanded coverage with 18 parameterized cases covering a broad range of azimuth, elevation, and range values, including boundary and near-boundary scenarios and tiny offsets.

### CODE REVIEW

**Review by phc1990 — APPROVED** [review]

File: src/OpenSpaceToolkit/Astrodynamics/Access/VisibilityCriterion/AERMask.cpp [hunk 1]
```diff
@@ -88,17 +88,27 @@ bool VisibilityCriterion::AERMask::isSatisfied(
...
-    auto itLow = this->azimuthElevationMask.lower_bound(anAzimuth_Radians);
-    itLow--;
-    auto itUp = this->azimuthElevationMask.upper_bound(anAzimuth_Radians);
+    Real anAzimuthReduced_Radians = Angle(anAzimuth_Radians, Angle::Unit::Radian).inRadians(0.0, Real::TwoPi());
+    Real anElevationReduced_Radians =
+        Angle(anElevation_Radians, Angle::Unit::Radian).inRadians(-Real::Pi(), Real::Pi());
+
+    auto itLow = this->azimuthElevationMask.lower_bound(anAzimuthReduced_Radians);
+
+    if (itLow->first != anAzimuthReduced_Radians)
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Access/VisibilityCriterion/AERMask.cpp:97]: Yeah this is okay as we are checking for exact values from the map.

File: test/OpenSpaceToolkit/Astrodynamics/Access/VisibilityCriterion.test.cpp [hunk 2]
```diff
@@ -425,28 +425,90 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Access_VisibilityCriterion, AERIntervalIsS
...
+    EXPECT_EQ(
+        aerMask.value().isSatisfied(param.azimuth.inRadians(), param.elevation.inRadians(), param.range.inMeters()),
+        param.expectedResult
+    );
+
+    if (std::abs(param.elevation.inDegrees()) <= 90.0 && param.azimuth.inDegrees() <= 360.0)
+    {
+        const AER aer(param.azimuth, param.elevation, param.range);
+        EXPECT_EQ(aerMask.value().isSatisfied(aer), param.expectedResult);
     }
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Access/VisibilityCriterion.test.cpp:474]: It's just checking both interfaces to the function are working as expected. I think it's fine.
